pkg/controllers/federate/controller.go

```diff
@@ -272,6 +273,37 @@ func (c *FederateController) reconcile(ctx context.Context, key workerKey) (stat
 		fedObject = fedObject.DeepCopyGenericFederatedObject()
 	}
 
+	if fedObject != nil {
+		// To account for the very small chance of name collision, we verify the owner reference before proceeding.
+		ownedbySource := false
+		requiresUpdate := -1
```

Comment:
We don't need to set the OR here. The update logic should already take care of that.

Reply:
Fixed